2 dz/3/main.cpp

```diff
@@ -0,0 +1,37 @@
+/*Напечатать все представления натурального числа N суммой натуральных слагаемых. Перестановка слагаемых нового способа не дает.*/
+
+#include <iostream>
+
+using namespace std;
+
+void the_executor(int *a, int n1, int pos, int N)
+{
+    for (int i = n1; i <= N / 2; i++)
+    {
+        a[pos] = i;
+        the_executor(a, i, pos + 1, N - i);
+    }
+    for (int i = 0; i < pos; i++)
+    {
+        cout << a[i] << " + ";
+    }
+    cout << N;
+    cout << endl;
+}
+
+void representation_of_number(int N)
+{
+    int arr_of_term[150] = {};
```

Comment:
Magic constant. Maybe use "150" as a default argument of the function?
Style guide p. 12
https://en.cppreference.com/w/cpp/language/default_arguments